pages/signin.page.ts

```diff
@@ -0,0 +1,40 @@
+import { Page } from "@playwright/test";
+import { commonData } from "../Data/common.data";
+import { signinPageData } from "../Data/signinPage.data";
+import { stateEnum } from "../enums/state.enum";
+import { factory } from "../factory";
+import { mainPageSelectors } from "../selectors/mainPage.selectors";
+import {signinPageSelectors} from "../selectors/signinPage.selectors";
+import { utils } from "../Utils/utils";
+
+export default class signinPage
+{
+    private readonly utils : utils;
+
+    constructor(page : Page)
+    {
+        this.utils = factory.initUtils(page);
+    }
+
+    async gotoSigninPage() : Promise<void>
+    {
+        await this.utils.goto(commonData.gitHubUrl + signinPageData.signinUrl, signinPageSelectors.emailSelector);
+    }
+
+    async signin(email : string, password : string, saveContext : boolean = true) : Promise<void>
```

Comment:
why do you need to pass saveContext here, the function should always same the storage state as long as its called, which means that its not called unless there is no storage state